File: PASCAL/Cap06/c06ex03.pas
program c06ex03;

type
  TAluno = Record
    Nome: String;
    Turma: Char;
    Sala: Integer;
    Notas: array[1..4] of Real;
  end;

var
  ALUNO: array[1..8] of TAluno;
  I, J: Integer;

begin
  WriteLn('Escola de Computacao XPTO');
  WriteLn('Cadastro Escolar de Aluno');
  WriteLn;

  for I := 1 to 8 do
    begin
      WriteLn('Aluno: ', I);
      WriteLn;

      Write('Nome ..............: ');
      ReadLn(ALUNO[I].Nome);

      Write('Turma .............: ');
      ReadLn(ALUNO[I].Turma);

      Write('Sala ..............: ');
      ReadLn(ALUNO[I].Sala);
      WriteLn;

      for J := 1 to 4 do
        begin
          Write('Entre a ', J, 'a. nota ..: ');
          ReadLn(ALUNO[I].Notas[J]);
        end;

      WriteLn;
    end;

  WriteLn('Escola de Computacao XPTO');
  WriteLn('Cadastro Escolar de Aluno');
  WriteLn;
  WriteLn('DADOS DOS ALUNOS');
  Write('Aluno ');
  Write('Nome                           ');
  Write('Sala ');
  Write('Turma ');
  Write('Nota1 ');
  Write('Nota2 ');
  Write('Nota3 ');
  Write('Nota4 ');
  WriteLn;
  Write('----- ');
  Write('------------------------------ ');
  Write('---- ');
  Write('----- ');
  Write('----- ');
  Write('----- ');
  Write('----- ');
  Write('----- ');
  WriteLn;

  for I := 1 to 8 do
    begin
      Write(I:5, ' ');
      Write(ALUNO[I].Nome:30, ' ');
      Write(ALUNO[I].Sala:4, ' ');
      Write(ALUNO[I].Turma:5, ' ');
      for J := 1 to 4 do
        Write(ALUNO[I].Notas[J]:5:1, ' ');
      WriteLn;
    end;

  WriteLn;
  Write('Tecle <Enter> para encerrar... ');
  ReadLn;
end.

File: C/Cap06/c06ex03.c
// c06ex03.c

#include <stdio.h>

typedef struct {
  char Nome[50];
  char Turma;
  int Sala;
  float Notas[4];
} Aluno;

int main(void) {
  Aluno ALUNO[8];
  int I, J;

  printf("Escola de Computacao XPTO\n");
  printf("Cadastro Escolar de Aluno\n\n");

  for (I = 0; I <= 7; ++I) {
    printf("Aluno: %d\n\n", I + 1);

    printf("Nome ..............: ");
    scanf("%s", ALUNO[I].Nome);
    while (getchar() != '\n');

    printf("Turma .............: ");
    scanf(" %c", &ALUNO[I].Turma);
    while (getchar() != '\n');

    printf("Sala ..............: ");
    scanf("%d", &ALUNO[I].Sala);
    while (getchar() != '\n');
    printf("\n");

    for (J = 0; J <= 3; ++J) {
      printf("Entre a %da. nota ..: ", J + 1);
      scanf("%f", &ALUNO[I].Notas[J]);
      while (getchar() != '\n');
   }

    printf("\n");
  }

  printf("Escola de Computacao XPTO\n");
  printf("Cadastro Escolar de Aluno\n\n");
  printf("DADOS DOS ALUNOS\n");

  printf("Aluno ");
  printf("Nome                           ");
  printf("Sala ");
  printf("Turma ");
  printf("Nota1 ");
  printf("Nota2 ");
  printf("Nota3 ");
  printf("Nota4\n");

  printf("----- ");
  printf("------------------------------ ");
  printf("---- ");
  printf("----- ");
  printf("----- ");
  printf("----- ");
  printf("----- ");
  printf("-----\n");

  for (I = 0; I <= 7; ++I) {
    printf("%5d ", I + 1);
    printf("%-30s ", ALUNO[I].Nome);
    printf("%4d ", ALUNO[I].Sala);
    printf("%5c ", ALUNO[I].Turma);
    for (J = 0; J <= 3; ++J) {
      printf("%5.1f ", ALUNO[I].Notas[J]);
    }
    printf("\n");
  }

  printf("\n");
  printf("Tecle <Enter> para encerrar... ");
  while (getchar() != '\n');

  return 0;
}
